From 7c5cad61d72f76d55e5d49ba412c94823a1a6113 Mon Sep 17 00:00:00 2001
From: Abdellatif El Khlifi <abdellatif.elkhlifi@arm.com>
Date: Tue, 2 Nov 2021 16:44:39 +0000
Subject: [PATCH 10/27] arm_ffa: corstone1000: enable FF-A and MM support

This commit allows corstone1000 platform to perform
MM communication between u-boot and the secure world
using FF-A transport.

Signed-off-by: Abdellatif El Khlifi <abdellatif.elkhlifi@arm.com>
Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
---
 configs/corstone1000_defconfig |  1 +
 include/configs/corstone1000.h | 15 +++++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/configs/corstone1000_defconfig b/configs/corstone1000_defconfig
index e573fe6fe6a2..b042d4e49419 100644
--- a/configs/corstone1000_defconfig
+++ b/configs/corstone1000_defconfig
@@ -44,6 +44,7 @@ CONFIG_USB=y
 CONFIG_DM_USB=y
 CONFIG_USB_ISP1760=y
 CONFIG_USB_STORAGE=y
+CONFIG_ARM_FFA_TRANSPORT=y
 CONFIG_EFI_MM_COMM_TEE=y
 # CONFIG_OPTEE is not set
 # CONFIG_GENERATE_SMBIOS_TABLE is not set
diff --git a/include/configs/corstone1000.h b/include/configs/corstone1000.h
index 8ba0effb0ab2..afc9ccfc192b 100644
--- a/include/configs/corstone1000.h
+++ b/include/configs/corstone1000.h
@@ -14,6 +14,21 @@
 
 #include <linux/sizes.h>
 
+/* MM SP UUID binary data (little-endian format) */
+#define MM_SP_UUID_DATA	\
+	0xed, 0x32, 0xd5, 0x33,	\
+	0x99, 0xe6, 0x42, 0x09,	\
+	0x9c, 0xc0, 0x2d, 0x72,	\
+	0xcd, 0xd9, 0x98, 0xa7
+
+#define FFA_SHARED_MM_BUFFER_SIZE	SZ_4K /* 4 KB */
+
+/*
+ * shared buffer physical address used for communication between
+ * u-boot and the MM SP
+ */
+#define FFA_SHARED_MM_BUFFER_ADDR	(0x023F8000)
+
 #define CONFIG_SYS_INIT_SP_ADDR		(CONFIG_SYS_SDRAM_BASE + 0x03f00000)
 #define CONFIG_SKIP_LOWLEVEL_INIT
 
-- 
2.35.1

